meson.build

```diff
@@ -54,7 +55,7 @@ libnvme_mi_dep = dependency('libnvme-mi', required: true,
 if get_option('json-c').disabled()
     json_c_dep = dependency('', required: false)
 else
-    json_c_dep = dependency('json-c', required: true, version: '>=0.13',
+    json_c_dep = dependency('json-c', required: get_option('json-c'), version: '>=0.13',
                             fallback : ['json-c', 'json_c_dep'])
     if json_c_dep.version().version_compare('>=0.14')
```

Comment:
If json-c is no longer mandatory then you should be checking that before inserting `requires = 'Requires: json-c >= 0.13'` in the case where it is not found.

Reply:
Right, but what happens if it is the value "auto" and the library isn't present?

Reply:
That idea is that should be treated like disabled.

Background some members of the community have expressed that they dislike all the newly introduced dependencies, that's why I am trying to cut them down. Obviously, the feature set is also cut down, not much I can do there.

Reply:
> That idea is that should be treated like disabled.

This is not how meson treats anything at all. Instead, meson treats disabled like "not found".

And programmatically what you get is that `json_c_dep.found() == false` and `json_c_dep.version() == 'unknown'`. Your above check for `get_option('json-c').disabled()` doesn't match because it is possible to get "not found" without "disabled".

...

Personally I'd tend to think it makes more sense to not manually check this. It's too complicated and meson can do it builtin.

```meson
json_c_dep = dependency('json-c', required: get_option('json-c'), version: '>=0.13',
                        fallback : ['json-c', 'json_c_dep'])
if json_c_dep.found()
    if json_c_dep.version().version_compare('>=0.14')
    .........
```

Why do you need to open-code your own disabled handling?